### ISSUE TITLE
Research how to pretrain Decoder language model

### ISSUE THREAD

**Issue by new5558** [OpenThaiGPT/openthaigpt-pretraining#28]
Example: Prelix LM, Loss Function
Objective: Find alternative pretraining decoder LLM that may be better than standard autoregressive

**Comment by new5558** [comment]
We will do Autoregressive for now
